### PR TITLE
Add `uv pip check`

### PR DESCRIPTION
**Pull request by ChannyClaus**
## Summary
Resolves https://github.com/astral-sh/uv/issues/2391

## Test Plan
Added a few tests to make sure that the exit code returned is 0 when there's no conflict; 1 when there's any conflict.

### CODE REVIEW

**Comment by danielhollas** [comment]
Thanks for taking this on @ChannyClaus, much appreciated! :love_you_gesture:

File: crates/uv/tests/pip_check.rs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+        .arg("check")
+        .arg("--cache-dir")
+        .arg(context.cache_dir.path())
+        .env("VIRTUAL_ENV", context.venv.as_os_str())
+        .current_dir(&context.temp_dir), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    No broken requirements found.
+
```
Comment (zanieb) [inline crates/uv/tests/pip_check.rs:75]: Is this an extra newline being omitted?

File: crates/uv/src/commands/pip_check.rs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    for diagnostic in site_packages.diagnostics()? {
+        is_compatible = false;
+        writeln!(printer.stdout(), "{}", diagnostic.message())?;
+    }
+
+    if !is_compatible {
+        return Ok(ExitStatus::Failure);
+    }
+
+    writeln!(printer.stdout(), "No broken requirements found.").unwrap();
```
Comment (zanieb) [inline crates/uv/src/commands/pip_check.rs:59]: Could we use a positive message instead? e.g. "Installed packages passed checks"

File: crates/uv/tests/pip_check.rs (hunk 1 above)
Comment (ChannyClaus) [reply crates/uv/tests/pip_check.rs:75]:
```
chankang@chans-MacBook-Air ~/repos/uv -  (uvcheck)
$ cargo run --quiet pip check
No broken requirements found.
chankang@chans-MacBook-Air ~/repos/uv -  (uvcheck)
$ 
```

```
chankang@chans-MacBook-Air ~/repos/uv -  (uvcheck)
$ target/debug/uv pip check
No broken requirements found.
chankang@chans-MacBook-Air ~/repos/uv -  (uvcheck)
$ 
```

_I think_ it's just one

File: crates/uv/src/commands/pip_check.rs (hunk 2 above)
Comment (ChannyClaus) [reply crates/uv/src/commands/pip_check.rs:59]: The intention was to match `pip`s' behavior - can still make the change if the positive one is preferred!
```
$ pip check
No broken requirements found.
```

Comment (zanieb) [reply crates/uv/src/commands/pip_check.rs:59]: I don't think we need to match their output (we're not elsewhere) and I have a minor preference for happy sounding success messages haha

Comment (zanieb) [reply crates/uv/src/commands/pip_check.rs:59]: Thanks for considering that though!

Comment (ChannyClaus) [reply crates/uv/src/commands/pip_check.rs]: changed it to `Installed packages pass the check.` 👍

**Comment by ChannyClaus** [comment]
hmm pushed a benign-looking commit and CI is suddenly very unhappy 😭 - fixing

**Review by zanieb — APPROVED** [review]

**Pull request merged**